test/support/integration/enterprise_server.ex

```diff
@@ -0,0 +1,186 @@
+        {term, collectable_fun} = Collectable.into(collectable)
+        {term, wrap(collectable_fun, fun)}
+      end
+
+      defp wrap(collectable_fun, fun) do
+        fn
+          term, {:cont, x} ->
+            collectable_fun.(term, {:cont, fun.(x)})
+
+          term, command ->
+            collectable_fun.(term, command)
+        end
+      end
+    end
+  end
+
+  def start_link do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def url do
+    GenServer.call(__MODULE__, :fetch_url)
+  end
+
+  def token do
+    GenServer.call(__MODULE__, :fetch_token)
+  end
+
+  # GenServer Callbacks
+
+  @impl true
+  def init(_opts) do
+    {:ok, %{token: nil, url: nil, port: nil}, {:continue, :start_enterprise}}
+  end
+
+  @impl true
+  def handle_continue(:start_enterprise, state) do
+    {:noreply, %{state | port: start_enterprise(), url: "http://localhost:4043"}}
+  end
+
+  @impl true
+  def handle_call(:fetch_token, _from, state) do
+    token = state.token || fetch_token_from_enterprise()
+
+    {:reply, token, %{state | token: token}}
+  end
+
+  def handle_call(:fetch_url, _from, state) do
+    {:reply, state.url, state}
+  end
+
+  # Port Callbacks
+
+  @impl true
+  def handle_info({_port, {:data, msg}}, state) do
+    [:blue, msg]
+    |> IO.ANSI.format()
+    |> IO.write()
+
+    {:noreply, state}
+  end
+
+  def handle_info({_port, {:exit_status, status}}, _state) do
+    [:red, "enterprise quit with status #{status}"]
+    |> IO.ANSI.format()
+    |> IO.write()
+
+    System.halt(status)
+  end
+
+  # Private
+
+  defp fetch_token_from_enterprise do
+    token = cmd(["enterprise.gen.token"], with_return: true)
+    String.trim(token)
+  end
+
+  defp start_enterprise do
+    ensure_app_dir!()
+    prepare_database()
+
+    executable = :os.find_executable(~c"elixir")
+    env = [{~c"MIX_ENV", ~c"livebook"}]
+
+    args = [
+      "-e",
+      "spawn(fn -> IO.gets([]) && System.halt(0) end)",
+      "-S",
+      "mix",
+      "phx.server"
+    ]
+
+    port =
+      Port.open({:spawn_executable, executable}, [
+        :exit_status,
+        :use_stdio,
+        :stderr_to_stdout,
+        :binary,
+        :hide,
+        env: env,
+        cd: app_dir(),
+        args: args
+      ])
+
+    wait_on_start(port)
+  end
+
+  defp prepare_database do
+    cmd(["ecto.drop", "--quiet"])
+    cmd(["ecto.create", "--quiet"])
+    cmd(["ecto.migrate", "--quiet"])
+  end
+
+  defp ensure_app_dir! do
+    dir = app_dir()
+
+    unless File.exists?(dir) do
+      IO.puts(
+        "Unable to find #{dir}, make sure to clone the enterprise repository " <>
+          "into it to run integration tests or set ENTERPRISE_PATH to its location"
+      )
+
+      System.halt(1)
+    end
+  end
+
+  defp app_dir do
+    System.get_env("ENTERPRISE_PATH") || "../enterprise"
+  end
+
+  defp wait_on_start(port) do
+    case :httpc.request(:get, {~c"http://localhost:4043/public/health", []}, [], []) do
```

Comment:
Let's extract the port to a module variable :)

Reply:
Works too, with that we could allow something like `LIVEBOOK_ENTERPRISE_PORT=0` to make it start on a random port, if we want.

Reply:
Yeah, this works pretty well too

Reply:
> Works too, with that we could allow something like `LIVEBOOK_ENTERPRISE_PORT=0` to make it start on a random port, if we want.

Since we remove every `LIVEBOOK_` prefixed env var, I've chaged to `ENTERPRISE_PORT`, but I'm sending as `LIVEBOOK_ENTERPRISE_PORT` to enterprise application